### PR TITLE
8287771: Remove useless G1 After GC summary refinement and sampling thread times

### PR DESCRIPTION
**Pull request by tqxia**
In the "After GC summary" printed by gc+remset=trace and G1RemsetStatsSummaryPeriod > 1 we print refinement and sampling thread times. However during GC neither thread types are active, so it always prints zeros.

Remove the printing of these unnecessary values in the after gc summary.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8287771](https://bugs.openjdk.org/browse/JDK-8287771): Remove useless G1 After GC summary refinement and sampling thread times

### Reviewers
 * [Thomas Schatzl](https://openjdk.org/census#tschatzl) (@tschatzl - **Reviewer**)
 * [Kim Barrett](https://openjdk.org/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to [a2c7dde2](https://git.openjdk.org/jdk/pull/9106/files/a2c7dde2d40adb53e957c7342247c3538fccb4a8)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9106/head:pull/9106` \
`$ git checkout pull/9106`

Update a local copy of the PR: \
`$ git checkout pull/9106` \
`$ git pull https://git.openjdk.org/jdk pull/9106/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9106`

View PR using the GUI difftool: \
`$ git pr show -t 9106`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9106.diff">https://git.openjdk.org/jdk/pull/9106.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/g1/g1RemSet.cpp [hunk 1]
```diff
@@ -1763,7 +1763,7 @@ void G1RemSet::enqueue_for_reprocessing(CardValue* card_ptr) {
   dcqs.enqueue_completed_buffer(BufferNode::make_node_from_buffer(buffer, index));
 }
 
-void G1RemSet::print_periodic_summary_info(const char* header, uint period_count) {
+void G1RemSet::print_periodic_summary_info(const char* header, uint period_count, bool include_vtimes) {
```
Comment (tschatzl) [inline src/hotspot/share/gc/g1/g1RemSet.cpp:1766]:
```suggestion
void G1RemSet::print_periodic_summary_info(const char* header, uint period_count, bool show_thread_times) {
```

I think the suggested parameter name is more clear than the original one.

**Review by tschatzl — CHANGES_REQUESTED** [review]
Looks good apart from the parameter name.

**Review by kimbarrett — APPROVED** [review]
Looks good, apart from the parameter name, where I agree with @tschatzl .

**Comment by kimbarrett** [comment]
An alternative to the new bool argument to print_on would be to add a new
function print_vtimes_on, and call it in the appropriate places before calling
print_on.  Just tossing the idea out there.

File: src/hotspot/share/gc/g1/g1CollectedHeap.cpp [hunk 2]
```diff
@@ -2803,7 +2803,7 @@ bool G1CollectedHeap::do_collection_pause_at_safepoint(double target_pause_time_
 
 G1HeapPrinterMark::G1HeapPrinterMark(G1CollectedHeap* g1h) : _g1h(g1h), _heap_transition(g1h) {
   // This summary needs to be printed before incrementing total collections.
-  _g1h->rem_set()->print_periodic_summary_info("Before GC RS summary", _g1h->total_collections());
+  _g1h->rem_set()->print_periodic_summary_info("Before GC RS summary", _g1h->total_collections(), true);
```
[the remark this answers is not in the document]
Comment (tqxia) [reply src/hotspot/share/gc/g1/g1CollectedHeap.cpp]: Got it.

File: src/hotspot/share/gc/g1/g1RemSet.cpp (hunk 1 above)
Comment (tqxia) [reply src/hotspot/share/gc/g1/g1RemSet.cpp]: Updated.

**Comment by tqxia** [comment]
> An alternative to the new bool argument to print_on would be to add a new function print_vtimes_on, and call it in the appropriate places before calling print_on. Just tossing the idea out there.

Thanks for the suggestion. After playing with the code a little bit more, I guess I will stick with the new flag for now for 2 reasons:
1. There is some shared calculation before printing thread times and printing other rem set information.
2. Since print_on(or print_thread_times_on) will be a method on G1RemSetSummary, in order to invoke it in G1HeapPrinterMark(where the decision is made), we probably need to create another wrapper for it on G1RemSet.

**Review by tschatzl — APPROVED** [review]

**Comment by tqxia** [comment]
Thanks for the review!

/integrate

**Pull request closed without merging**